javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -459,6 +475,28 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         }
     }
 
+    /**
+     * Emits the constructor parameters for a record type. i.e. {@code (String name, int count)}.
+     */
+    private void emitRecordConstructor(CodeWriter codeWriter) throws IOException {
+        codeWriter.emit("(");
+        // Only include non-static fields in the constructor.
+        List<FieldSpec> nonStaticFields = fieldSpecs.stream()
+                .filter(field -> !field.modifiers().contains(Modifier.STATIC))
+                .toList();
```

Comment:
I don't think we should model record components as fields. There's a lot of stuff in a `FieldSpec` that is simply not relevant for record components. It's weird to simply ignore the irrelevant bits and then have logic like this that special-cases certain kinds of fields.

Reply:
Ah I didn't realise there was a PR already up for this... I have completely just ripped off their PR...